components/Toggle/Togglev2.jsx

```diff
@@ -0,0 +1,223 @@
+		secondary,
+	} = props;
+	const [sliderLeft, setSliderLeft] = useState(0);
+	const [sliderWidth, setSliderWidth] = useState(0);
+
+	const sliderRef = useRef(null);
+
+	const updateSliderPosition = () => {
+		const activeTabElement = sliderRef.current;
+		if (activeTabElement) {
+			setSliderLeft(activeTabElement.offsetLeft);
+			setSliderWidth(activeTabElement.offsetWidth);
+		}
+	};
+
+	// for uncontrolled input
+	const [uncontrolledValue, setUncontrolledValue] = useState(() => {
+		if (multi) {
+			return defaultValue ?? options?.[0]?.value ?? [];
+		}
+		return defaultValue ?? options?.[0]?.value;
+	});
+
+	const { current: isControlled } = useRef(value !== undefined);
+
+	// eslint-disable-next-line no-nested-ternary
+	const inputValue = isControlled ? value ?? '' : uncontrolledValue;
+
+	const allSelected = multi ? options.length === inputValue.length : false;
+
+	const onSelectAll = () => {
+		const allValues = options.map((option) => {
+			return option.value;
+		});
+		if (isControlled) {
+			onChange(allValues);
+		} else {
+			setUncontrolledValue(allValues);
+		}
+	};
+
+	const onButtonClick = (newValue, selected) => {
+		// if multi select
+		if (multi) {
+			// if it was already selected, remove it
+			if (selected) {
+				const newInputValue = inputValue.filter((val) => {
+					return val !== newValue;
+				});
+
+				// if total selected after removal is greater than 0, set that
+				// else select all
+				if (newInputValue.length > 0) {
+					if (isControlled) {
+						onChange(newInputValue);
+					} else {
+						setUncontrolledValue(newInputValue);
+					}
+				} else {
+					onSelectAll();
+				}
+			} else {
+				// if all are selected, select only the one being clicked
+				if (allSelected) {
+					const newInputValue = [newValue];
+					if (isControlled) {
+						onChange(newInputValue);
+					} else {
+						setUncontrolledValue(newInputValue);
+					}
+				} else {
+					const newInputValue = [...inputValue, newValue];
+					if (isControlled) {
+						onChange(newInputValue);
+					} else {
+						setUncontrolledValue(newInputValue);
+					}
+				}
+			}
+		} else {
+			// if single select, just select that item
+			if (isControlled) {
+				onChange(newValue);
+			} else {
+				setUncontrolledValue(newValue);
+			}
+		}
+	};
+
+	const compareSelection = (input, item) => {
+		if (Array.isArray(input)) {
+			return input?.includes(item);
+		}
+		return input === item;
+	};
+	if (loading || fallback) {
+		return <Skeleton theme={theme} fallback={!loading && fallback} v2 />;
+	}
+
+	useEffect(() => {
+		updateSliderPosition();
+	}, [inputValue, options]);
+
+	return (
+		<div className={classes(styles.root, styles[theme], className)}>
+			<div className={styles.toggle}>
+				{multi && (
+					<Button
+						type='button'
+						size='auto'
+						data-elem='toggle'
+						variant='text'
+						className={classes(
+							styles['toggle-button'],
+							// eslint-disable-next-line no-nested-ternary
+							allSelected
```

Comment:
Try to convert this ternary condition in the same way as [this one](https://github.com/banyandata-cloud/roots/pull/624/files?file-filters%5B%5D=.jsx&file-filters%5B%5D=.scss&show-viewed-files=true#r1700621744)